homeassistant/components/acmeda/__init__.py

```diff
@@ -28,6 +32,19 @@ async def async_setup_entry(
     return True
 
 
+async def _migrate_unique_ids(hass: HomeAssistant, entry: AcmedaConfigEntry) -> None:
+    """Migrate pre-config flow unique ids."""
+    entity_registry = er.async_get(hass)
+    registry_entries = er.async_entries_for_config_entry(
+        entity_registry, entry.entry_id
+    )
+    for reg_entry in registry_entries:
+        if isinstance(reg_entry.unique_id, int):
```

Comment:
Mypy is correct here, as there should never have been an `int` in the first place.
We need to bypass mypy in this scenario:
```suggestion
        if isinstance(reg_entry.unique_id, int):   # type: ignore[unreachable]
```